File: src/SSHServer/Ciphers/NoCipher.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace SSHServer.Ciphers
{
    public class NoCipher : ICipher
    {
        public uint BlockSize
        {
            get
            {
                return 8;
            }
        }

        public uint KeySize
        {
            get
            {
                return 0;
            }
        }

        public string Name
        {
            get
            {
                return "none";
            }
        }

        public byte[] Decrypt(byte[] data)
        {
            return data;
        }

        public byte[] Encrypt(byte[] data)
        {
            return data;
        }

        public void SetKey(byte[] key, byte[] iv)
        {
            // No key for this cipher
        }
    }
}

File: SSHServerNodeJS/SSHServerNodeJS/Ciphers/NoCipher.ts
import { ICipher } from "./ICipher";

export class NoCipher implements ICipher {
    public getName(): string {
        return "none";
    }

    public getBlockSize(): number {
        return 8;
    }

    public getKeySize(): number {
        return 0;
    }

    public encrypt(data: Buffer): Buffer {
        return data;
    }

    public decrypt(data: Buffer): Buffer {
        return data;
    }

    public setKey(key: Buffer, iv: Buffer): void {
    }
}
